### PR TITLE
list more than 100 kv namespaces

### PR DESCRIPTION
**Pull request by ags799**
Use the KV API's pagination to accumulate all the namespaces in an account.

### CODE REVIEW

File: src/kv/namespace/upsert.rs [hunk 1]
```diff
@@ -31,7 +31,6 @@ pub fn upsert(
                     log::info!("Namespace {} already exists.", title);
 
                     let namespace = list(&client, target)?
-                        .result
                         .iter()
                         .find(|ns| ns.title == title)
                         .unwrap()
```
Comment (koeninger) [inline src/kv/namespace/upsert.rs:36]: Is this the unwrap that was panicking originally?  Should we go ahead and fix it to return an error, even if it "shouldn't" happen anymore?

File: src/kv/namespace/list.rs [hunk 2]
```diff
@@ -4,24 +4,50 @@ use cloudflare::endpoints::workerskv::list_namespaces::ListNamespaces;
...
+            }
+            Err(e) => failure::bail!("{}", kv::format_error(e)),
+        }
+    }
+    Ok(namespaces)
+}
+
+fn get_total(list_response: &ApiSuccess<Vec<WorkersKvNamespace>>) -> Result<usize, failure::Error> {
+    let result_info: ListResponseResultInfo =
+        serde_json::from_value(list_response.result_info.clone().unwrap())?;
```
Comment (koeninger) [inline src/kv/namespace/list.rs:46]: I'd like to get out of the habit of bare unwraps in the codebase, especially when this function is already returning a result.

File: src/kv/namespace/upsert.rs (hunk 1 above)
Comment (ags799) [reply src/kv/namespace/upsert.rs]: done

File: src/kv/namespace/list.rs (hunk 2 above)
Comment (ags799) [reply src/kv/namespace/list.rs]: done

**Review by koeninger — APPROVED** [review]

File: src/kv/namespace/upsert.rs [hunk 3]
```diff
@@ -30,13 +30,12 @@ pub fn upsert(
...
-                    let namespace = list(&client, target)?
-                        .result
+                    match list(&client, target)?
                         .iter()
-                        .find(|ns| ns.title == title)
-                        .unwrap()
-                        .to_owned();
-                    Ok(UpsertedNamespace::Reused(namespace))
+                        .find(|ns| ns.title == title) {
+                        Some(namespace) => Ok(UpsertedNamespace::Reused(namespace.to_owned())),
```
Comment (xortive) [inline src/kv/namespace/upsert.rs:36]: :smile:

**Review by xortive — APPROVED** [review]

**Pull request merged**